tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/FindReplaceDialogTest.java

```diff
@@ -366,6 +378,37 @@ public void testFindWithWholeWordEnabledWithMultipleWords() {
 		assertEquals(dialog.findCombo.getText().length(), (target.getSelection()).y);
 	}
 
+	@Test
+	public void testGlobalSearchChangeScope() {
```

Comment:
Shouldn't this be a test case of the `FindReplaceLogic` rather than (or in addition to) the `FindReplaceDialog`? It already affects the business logic, no matter which UI is used to execute it.

Reply:
I have removed the test and added a similar test in FindReplaceLogicTest with this commit: https://github.com/eclipse-platform/eclipse.platform.ui/commit/fa4cfb1121b8ac0f2fe8f1bbb0ae46cc656dc22e